src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/mailer.html.twig

```diff
@@ -139,28 +139,31 @@
                                                 {# Email instance #}
-                                                <div class="tab">
-                                                    <h3 class="tab-title">HTML preview</h3>
-                                                    <div class="tab-content">
-                                                        <pre class="prewrap" style="max-height: 600px">
-                                                            <iframe
-                                                                src="data:text/html;base64;charset=utf-8,{{ collector.base64Encode(message.htmlBody()) }}"
-                                                                style="height: 80vh;width: 100%;"
-                                                            >
-                                                            </iframe>
-                                                        </pre>
+                                                {% set htmlBody = message.htmlBody() %}
+                                                {% if htmlBody is not null %}
+                                                    <div class="tab">
+                                                        <h3 class="tab-title">HTML Preview</h3>
+                                                        <div class="tab-content">
+                                                            <pre class="prewrap" style="max-height: 600px">
+                                                                <iframe
+                                                                    src="data:text/html;charset=utf-8;base64,{{ collector.base64Encode(htmlBody) }}"
+                                                                    style="height: 80vh;width: 100%;"
+                                                                >
+                                                                </iframe>
+                                                            </pre>
+                                                        </div>
                                                     </div>
-                                                </div>
+                                                {% endif %}
                                                 <div class="tab">
                                                     <h3 class="tab-title">HTML Content</h3>
                                                     <div class="tab-content">
                                                         <pre class="prewrap" style="max-height: 600px">
                                                             {%- if message.htmlCharset() %}
-                                                                {{- message.htmlBody()|convert_encoding('UTF-8', message.htmlCharset()) }}
+                                                                {{- htmlBody|convert_encoding('UTF-8', message.htmlCharset()) }}
```

Comment:
It looks like the HTML Content tab has the same problem with `resource` because Twig filter `convert_encoding` uses `iconv()` internally.

Reply:
this is now wrapped inside an if htmlBody is not null
so it should be ok